pkg/controller/sessionsecret/session_secret_controller.go

```diff
@@ -0,0 +1,226 @@
+package sessionsecret
+
+import (
+	"fmt"
+	"time"
+
+	"github.com/golang/glog"
+
+	"k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/runtime/serializer"
+	utilruntime "k8s.io/apimachinery/pkg/util/runtime"
+	"k8s.io/apimachinery/pkg/util/wait"
+	informers "k8s.io/client-go/informers/core/v1"
+	kcoreclient "k8s.io/client-go/kubernetes/typed/core/v1"
+	listers "k8s.io/client-go/listers/core/v1"
+	"k8s.io/client-go/tools/cache"
+	"k8s.io/client-go/util/workqueue"
+
+	legacyv1 "github.com/openshift/api/legacyconfig/v1"
+	cryptohelpers "github.com/openshift/cluster-kube-apiserver-operator/pkg/operator/crypto"
+)
+
+const (
+	sessionSecretNamespace = "kube-system"
+	sessionSecretName      = "session-secret"
+)
+
+var (
+	scheme              = runtime.NewScheme()
+	codecs              = serializer.NewCodecFactory(scheme)
+	legacySchemeBuilder = runtime.NewSchemeBuilder(legacyv1.InstallLegacy)
+)
+
+func init() {
+	legacySchemeBuilder.AddToScheme(scheme)
+}
+
+type SessionSecretController struct {
+	secretClient     kcoreclient.SecretsGetter
+	secretLister     listers.SecretLister
+	secretsHasSynced cache.InformerSynced
+	syncHandler      func(serviceKey string) error
+	queue            workqueue.RateLimitingInterface
+}
+
+func NewSessionSecretController(secrets informers.SecretInformer, secretsClient kcoreclient.SecretsGetter, resyncInterval time.Duration) *SessionSecretController {
+	sc := &SessionSecretController{
+		queue: workqueue.NewRateLimitingQueue(workqueue.DefaultControllerRateLimiter()),
```

Comment:
nit: always give names to queues

Reply:
> nit: always give names to queues

required for monitoring.  Be sure to update.

Reply:
Will do.